rust/src/storage/azure/mod.rs

```diff
@@ -7,18 +7,28 @@
 /// `AZURE_STORAGE_ACCOUNT_NAME` is required to be set in the environment.
```

Comment:
nit: maybe remove the comment that the env is required, we have that also on top of the new method. Ideally metion the new alternative init approaches.

Reply:
Removed the top level comment since it was duplicate of the new comments. Updated it to mention alternative auth methods